src/shell/notification.js

```diff
@@ -252,50 +246,86 @@ const Source = GObject.registerClass({
                 : null;
 
             if (cachedNotification.title === title &&
-                cachedNotification.bannerBodyText === body) {
-                this._notificationPending = false;
-                return;
-            }
+                cachedNotification.body === body)
+                return cachedNotification;
 
             cachedNotification.title = title;
-            cachedNotification.bannerBodyText = body;
             cachedNotification.body = body;
 
-        // Device Notification
-        } else if (idMatch) {
-            cachedNotification = notification;
+            return cachedNotification;
+        }
 
-            cachedNotification.deviceId = deviceId;
-            cachedNotification.remoteId = remoteId;
-            cachedNotification.requestReplyId = requestReplyId;
+        // Device Notification
+        if (idMatch) {
+            notification.deviceId = deviceId;
+            notification.remoteId = remoteId;
+            notification.requestReplyId = requestReplyId;
 
-            cachedNotification.connect('destroy', (notification, reason) => {
+            notification.connect('destroy', (notification, reason) => {
                 this._closeGSConnectNotification(notification, reason);
                 delete this._notifications[localId];
             });
 
-            this._notifications[localId] = cachedNotification;
-
         // Service Notification
         } else {
-            cachedNotification.connect('destroy', (notification, reason) => {
+            notification.connect('destroy', (notification, reason) => {
                 delete this._notifications[localId];
             });
-            this._notifications[localId] = cachedNotification;
         }
 
-        if (this.notifications.includes(cachedNotification)) {
-            cachedNotification.acknowledged = false;
-            this.emit('notification-request-banner', cachedNotification);
+        this._notifications[localId] = notification;
+        return notification;
+    }
+
+    /*
+     * Override to control notification spawning
+     */
+    addNotification(notification) {
+        this._notificationPending = true;
+
+        // Fix themed icons
+        if (notification.icon) {
+            let gicon = notification.icon;
+
+            if (gicon instanceof Gio.ThemedIcon) {
+                gicon = getIcon(gicon.names[0]);
+                notification.icon = gicon.serialize();
+            }
+        }
+
+        const createdNotification = this._createNotification(notification);
+        this._addNotificationToMessageTray(createdNotification);
+
+        this._notificationPending = false;
+    }
+
+    /*
+     * Reimplementation of MessageTray.addNotification to raise the usual
+     * notification limit (3)
```

Comment:
(Belatedly, sorry... but...)

I wonder if this is a good idea? Just because, GNOME could _change_ the notification limit from `3` tomorrow, and then the comment's wrong. The more important point is probably that we raise it _to_ `10`.

I mean, it's just a comment, so it's not worth a PR just to change it anyway.